test/get-client.test.js

```diff
@@ -64,66 +70,73 @@ test.serial('Use a https proxy', async (t) => {
   const proxyPort = proxy.address().port;
   serverDestroy(proxy);
 
-  const proxyHandler = spy();
-  const serverHandler = spy((request, response) => {
+  const proxyHandler = sinon.spy();
+  const serverHandler = sinon.spy((request, response) => {
     response.end();
   });
-  proxy.on('connect', proxyHandler);
-  server.on('request', serverHandler);
+  proxy.on("connect", proxyHandler);
+  server.on("request", serverHandler);
 
   const github = getClient({
-    githubToken: 'github_token',
+    githubToken: "github_token",
     githubUrl: `https://localhost:${serverPort}`,
-    githubApiPathPrefix: '',
-    proxy: {host: 'localhost', port: proxyPort, headers: {foo: 'bar'}},
+    githubApiPathPrefix: "",
+    proxy: { host: "localhost", port: proxyPort, headers: { foo: "bar" } },
   });
 
-  await github.repos.get({repo: 'repo', owner: 'owner'});
+  await github.repos.get({ repo: "repo", owner: "owner" });
 
   t.is(proxyHandler.args[0][0].url, `localhost:${serverPort}`);
-  t.is(proxyHandler.args[0][0].headers.foo, 'bar');
-  t.is(serverHandler.args[0][0].headers.accept, 'application/vnd.github.v3+json');
+  t.is(proxyHandler.args[0][0].headers.foo, "bar");
+  t.is(serverHandler.args[0][0].headers.accept, "application/vnd.github.v3+json");
 
   await promisify(proxy.destroy).bind(proxy)();
   await promisify(server.destroy).bind(server)();
 });
 
-test.serial('Do not use a proxy if set to false', async (t) => {
-  const server = http.createServer();
+test.serial("Do not use a proxy if set to false", async (t) => {
+  const server = createServer();
   await promisify(server.listen).bind(server)();
   const serverPort = server.address().port;
   serverDestroy(server);
 
-  const serverHandler = spy((request, response) => {
+  const serverHandler = sinon.spy((request, response) => {
     response.end();
   });
-  server.on('request', serverHandler);
+  server.on("request", serverHandler);
 
   const github = getClient({
-    githubToken: 'github_token',
+    githubToken: "github_token",
     githubUrl: `http://localhost:${serverPort}`,
-    githubApiPathPrefix: '',
+    githubApiPathPrefix: "",
     proxy: false,
   });
 
-  await github.repos.get({repo: 'repo', owner: 'owner'});
+  await github.repos.get({ repo: "repo", owner: "owner" });
 
-  t.is(serverHandler.args[0][0].headers.accept, 'application/vnd.github.v3+json');
+  t.is(serverHandler.args[0][0].headers.accept, "application/vnd.github.v3+json");
   t.falsy(serverHandler.args[0][0].headers.via);
-  t.falsy(serverHandler.args[0][0].headers['x-forwarded-for']);
+  t.falsy(serverHandler.args[0][0].headers["x-forwarded-for"]);
 
   await promisify(server.destroy).bind(server)();
 });
 
-test('Use the global throttler for all endpoints', async (t) => {
+test.serial("Use the global throttler for all endpoints", async (t) => {
```

Comment:
what is the reason for needing serial on more of these tests? i'd prefer we work in the direction of not needing to run any tests serially since it usually means that there is bleed between tests. were there test problems that made these additions necessary?

Reply:
if it is necessary for now, i'm not worried enough about it to hold this up

Reply:
it was necessary for the tests to run. Same problem with latest `ava@3`. Doing `test.serial` instead of `test` made it all work. I agree with you though, I'd like to get rid of the `.serial` tests, too. I think it's a code smell